src/main/groovy/SlackRestHelper.groovy

```diff
@@ -0,0 +1,70 @@
+import org.apache.commons.httpclient.*
+import org.apache.commons.httpclient.methods.*
+import groovy.json.JsonSlurper
+
+public class SlackRestHelper {
+
+    def private sessionId
+    def private JsonSlurper jsonSlurper
+    def private HttpClient client
+    def private restUrl
+    def private baseUrl
+    def private cookie
+
+    def public SlackRestHelper() {
+
+        jsonSlurper = new JsonSlurper()
+        client = new HttpClient()
+    }
+
+    def private executePostMethod = { token, channel, message, username, as_user ->
+        HttpMethod method = new PostMethod("https://slack.com/api/chat.postMessage")
+        method.addRequestHeader("ContentType", "application/x-www-form-urlencoded")
+        method.addParameter("token", token)
+        method.addParameter("channel", channel)
+        method.addParameter("text", message)
+        method.addParameter("username", username)
+        method.addParameter("as_user", as_user)
+
+        def methodResult = client.executeMethod(method)
+        def response = method.getResponseBodyAsString()
+        def responseJson = null
+
+        if (methodResult == 200) {
+            responseJson = jsonSlurper.parseText(response)
+        }
+        else {
+            throw new Exception ("Received response ${methodResult}: ${response}")
+        }
+
+        if (responseJson.ok == false) {
+            throw new Exception ("Slack command error: " + responseJson.error)
+        }
+
+        println("Posted message successfully.")
+        return responseJson
+    }
+
+    def private executeIncomingWebhookPostMethod = {webhookURL, channel, message ->
```

Comment:
Is this method used anywhere?

Reply:
I think it would replace much of the logic here? 
https://github.com/UrbanCode/Slack-UCD/blob/master/src/main/zip/postToSlack.groovy

Options:
A) I'm okay with including this code, but you should likely comment it out or label its purpose so we know.
B) Update the code where you expect it to be used. (But likely should be done in a separate PR).

Reply:
New step that we have added, we are using only executePostMethod from SlackRestHelper.
Which i picked from slack ucd plugin (hcl github). We can remove it. Right now we are not using that method.